### DISCUSSION TITLE
Functions and aliases

### DISCUSSION THREAD

**Discussion by julianpinedayyz** [google/zx#555] [Help needed]
Hello. Really new to all this and would love some feedback.

Is it possible to register functions and aliases to be used globally?

I want to add a file to my .zshrc so It can be sourced with functions for now. Something like:

`tellDate(){ echo "Today's date is "${date}" }` (this is obviously super trivial in bash. The idea is to build more complex functions.)

And then after sourcing my .zshrc, I would be able to call the `tellDate` function like any other function made inside a bash file.

Any help would be greatly appreciated!

Thanks!

**Comment by antonmedv** [comment]
Nice idea 💡 
We can implement something like this in zx bin.

**Comment by julianpinedayyz** [comment]
That would be amazing! I mean, I don't know if it's redundant but for sure it would be great to create complex functions that can be run globally. I was thinking about implementing this with Fig's plugins but you still need to have the Bash knowledge to make it happen.
